### PR TITLE
Remove scapegoat from Marathon

### PR DESCRIPTION
**Pull request by timcharper**
I submit that Scapegoat provides less value than it costs to use. It is slow to run, produces lots of warnings, fails horribly with async/await (something that has been unaddressed for several years, leading to rampant SuppressWarnings statements). It is difficult to remember a time when it made a helpful suggestion at an error level. I submit does not catch the things that are actually important.

### CODE REVIEW

**Comment by timcharper** [comment]
See https://github.com/mesosphere/marathon/search?q=SuppressWarnings&unscoped_q=SuppressWarnings

**Comment by ichernetsky** [comment]
What about removing those `SuppressWarnings`?

**Comment by timcharper** [comment]
Removing SuppressWarnings will be good. Preference to get consensus first.

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -169,7 +169,6 @@ class MarathonSchedulerActor private (
       deploy(sender(), cmd)
 
     case cmd @ KillTasks(runSpecId, tasks) =>
-      @SuppressWarnings(Array("all")) /* async/await */
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: You rock! When I read the title I was wondering whether you removed there or not.

**Pull request merged**